arrow/arrow.go

```diff
@@ -22,14 +22,14 @@ type ArrowWriter struct {
 // must match. The number of rows in each chunk is determined by chunkSize.
 // The ArrowWriter will write to filePath.
 // This writing operation is threadsafe.
-func NewArrowWriter(filePath string, fieldNames []string, fieldTypes []arrow.DataType) (*ArrowWriter, error) {
+func NewArrowWriter(filePath string, fieldNames []string, fieldTypes []arrow.DataType, options []ipc.Option) (*ArrowWriter, error) {
```

Comment:
A more idiomatic way of passing options would be to make options variadic  - e.g., func NewArrowWriter( filePath string, fieldNames []string, fieldTypes []arrow.DataType, options **...ipc.Option**) - rather than passing a slice of options.

My overall suggestion is to consider approaching this by requiring the ipc.NewFileWriter to be passed to the NewArrowWriter as a pointer (or, as an option itself) if you want to be able alter it - you could have a fallback that uses the default settings. It's similar to how bytes.Buffer has a default buffer size that you can change. Also, I'd consider passing other elements, e.g. fieldNames as options. I think this approach helps me follow the logic and leaves room for future additions.